### PR TITLE
Experimental plugin api

### PR DESCRIPTION
**Pull request by Mm2PL**
Pull request checklist:

- [x] `CHANGELOG.md` was updated, if applicable

# Description

### CODE REVIEW

**Comment by Mm2PL** [comment]
- [ ] List plugins somewhere in the UI.

**Comment by Mm2PL** [comment]
Plugins go in `path/to/Chatterino_config/Plugins/`.

File: src/singletons/Plugins.cpp [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+    QDir pluginsDir = paths.pluginsDirectory;
+
+    const auto entryList = pluginsDir.entryList(QDir::Files);
+    for (const QString &fileName : entryList)
+    {
+        QPluginLoader loader(pluginsDir.absoluteFilePath(fileName));
+        QObject *plugin = loader.instance();
+        if (plugin)
+        {
+            qDebug() << "Loaded" << fileName;
```
Comment (zneix) [inline src/singletons/Plugins.cpp:29]: Use Qt-categorized logging (in all places in this file).

**Review by zneix — CHANGES_REQUESTED** [review]
Most nitpicks / suggested changes were mentioned on stream; A few of these that I can remember on top of my head are:
- Reduce the amount of hardcoded "magic strings / numbers"
- Make it impossible for plugins to access (read / write) settings outside of `/plugins` block


Also, the thing that obv doesn't work are the timeouts in http requests, these should be looked into.
And possibly, increase the timeout value on the `supinic.com` call from 5s, because it seems to exceed this value quite often (based on my testing). Maybe 10/15 seconds will cut it.

**Comment by Mm2PL** [comment]
Will rewrite for chaiscript (or lua whichever is less cancer to use)

**Pull request closed without merging**